### PR TITLE
Update mlflow with using resolve_tags

### LINKED ISSUE #6745: mlflow run context is not logged when using MLFlowLogger
## 🐛 Bug
When we use the basic mlflow logging via `with mlflow.start_run(): ...` context manager, we get a better supplementary info about the run (git commit sha, user, filename) rendered in the Tracking UI ([system tags](https://mlflow.org/docs/latest/tracking.html#system-tags))

But when we use `MLFlowLogger` as a logger in pytorch_lightning, this info is not logged. As a user, I'd like to have a mirrored functionality out-of-the-box.

I inspected the `start_run()` method of mlflow and deduced that the only thing is left while creating the run via MLflowClient is to add `resolve_tags` from the `context` package:
```python
# pytorch_lightning/loggers/mlflow.py
from mlflow.tracking.context.registry import resolve_tags
...
    def experiment(self) -> MLflowClient:
        if self._run_id is None:
            run = self._mlflow_client.create_run(experiment_id=self._experiment_id, tags=resolve_tags(self.tags))
```

I think it's a better idea to add those tags internally (meaning not to expect users doing that manually) as first - it's as seamless as in the default API, secondly - it's the pytorch_lightning that manages the mlflow's run anyways.

**PR is following ...**

### PR DESCRIPTION
**Pull request by stllfe**
Fixes #6745: adds additional info about the run, as in the native API. [This](https://github.com/PyTorchLightning/pytorch-lightning/issues/741#issuecomment-812028459) message is also related to what I'd like to solve.

### CODE REVIEW

File: pytorch_lightning/loggers/mlflow.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 try:
     import mlflow
     from mlflow.tracking import MlflowClient
+    from mlflow.tracking.context.registry import resolve_tags
```
Comment (Borda) [inline pytorch_lightning/loggers/mlflow.py:33]: seems this is not supported in older versions

Comment (stllfe) [reply pytorch_lightning/loggers/mlflow.py]: thanks! my bad haven't thought of it upfront. trying to fix it...

**Comment by stllfe** [comment]
Any suggestions on code formatting and typing checks?

**Review by awaelchli — APPROVED** [review]

**Comment by Borda** [comment]
> Any suggestions on code formatting and typing checks?

fixed, mind re-check whether it is what you want to get?

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/loggers/mlflow.py [hunk 2]
```diff
@@ -30,10 +30,24 @@
...
+    mlflow, MlflowClient, context = None, None, None
+
+
+if _MLFLOW_AVAILABLE:
+    # before v1.1.0
+    if hasattr(context, 'resolve_tags'):
+        from mlflow.tracking.context import resolve_tags
+    # since v1.1.0
+    elif hasattr(context, 'registry'):
+        from mlflow.tracking.context.resolve_tags import resolve_tags
```
Comment (stllfe) [inline pytorch_lightning/loggers/mlflow.py:46]: Oops, it's actually 
```python
from mlflow.tracking.context.registry import resolve_tags
```
With the fix above it works as intended!

**Review by stllfe — COMMENTED** [review]
In my testing, it shows that everything works great with at least `v1.0.0` and above of mlflow!

FYI, the part with `resolve_tags` works even below that version, but then some other places still do break. Say, in version `0.9.1` they had `run.info.run_uuid` as opposed to `run.info.run_id` in later versions, etc. However, there are more drastic changes in metrics logging API, so I guess it would be just easier to highlight somewhere that the lowest supported version is `v1.0.0`... In MLFlowLogger docs maybe?

File: pytorch_lightning/loggers/mlflow.py (hunk 2 above)
Comment (stllfe) [reply pytorch_lightning/loggers/mlflow.py]: Okay, now it looks like some other tests fail with that registry, gonna take a look...

**Comment by Borda** [comment]
@stllfe mind check the failing test?

**Comment by stllfe** [comment]
@Borda done! Removed that `_MLFLOW_AVAILABLE` so that `resolve_tags` would be defined either way. However, it seems there still are some failed tests, idk why though... Can you check it? I'm not that familiar with pytest :(

**Comment by carmocca** [comment]
> However, it seems there still are some failed tests, idk why though... Can you check it? I'm not that familiar with pytest :(

The failed tests are unrelated.

Can you also update the CHANGELOG?

Do you see a way to test this?

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**